src/privileged/autoplay/api.js

```diff
@@ -0,0 +1,332 @@
+"use strict";
+
+ChromeUtils.import("resource:///modules/PermissionUI.jsm");
+ChromeUtils.import("resource:///modules/SitePermissions.jsm");
+ChromeUtils.import("resource://gre/modules/ExtensionCommon.jsm");
+ChromeUtils.import("resource://gre/modules/PopupNotifications.jsm");
+ChromeUtils.import("resource://gre/modules/PrivateBrowsingUtils.jsm");
+ChromeUtils.import("resource://gre/modules/Services.jsm");
+ChromeUtils.import("resource://gre/modules/TelemetryController.jsm");
+
+const CID = ChromeUtils.import("resource://gre/modules/ClientID.jsm", {});
+const { EventManager } = ExtensionCommon;
+
+XPCOMUtils.defineLazyGetter(this, "gBrowserBundle", function() {
+  return Services.strings.createBundle("chrome://browser/locale/browser.properties");
+});
+
+function _once(target, name) {
+  const p = new Promise(function(resolve, reject) {
+    target.addEventListener(name, function() {
+      resolve();
+    }, {once: true});
+  });
+  return p;
+}
+
+function errorHandler(err) {
```

Comment:
It would probably be more idiomatic to use `Cu.reportError` instead of this. https://searchfox.org/mozilla-central/rev/83562422ecb0f5683da7a9a26ce05ce62bc0c882/js/xpconnect/idl/xpccomponents.idl#157

Reply:
OK, thank you!